src/actions/measurements.ts

```diff
@@ -517,14 +577,176 @@ export const changeMeasurementsDisplay = (
+  batch(() => {
+    dispatch({
+      type: CHANGE_MEASUREMENTS_GROUP_BY,
+      controls: newControls,
+      queryParams: createMeasurementsQueryFromControls(newControls, measurements.collectionToDisplay, measurements.defaultCollectionKey)
+    });
+
+    /* After the group by has been updated, update the measurement coloring data if needed */
+    updateMeasurementsColorData(
+      newControls.measurementsColorGrouping,
+      controls.measurementsColorGrouping,
+      controls.colorBy,
+      controls.defaults.colorBy,
+      dispatch
+    );
+  })
+}
+
+export function getActiveMeasurementFilters(
+  filters: MeasurementFilters
+): {string?: string[]} {
+  // Find active filters to filter measurements
+  const activeFilters: {string?: string[]} = {};
+  Object.entries(filters).forEach(([field, valuesMap]) => {
+    activeFilters[field] = activeFilters[field] || [];
+    valuesMap.forEach(({active}, fieldValue) => {
+      // Save array of active values for the field filter
+      if (active) activeFilters[field].push(fieldValue);
+    });
+  });
+  return activeFilters;
+}
+
+export function matchesAllActiveFilters(
+  measurement: Measurement,
+  activeFilters: {string?: string[]}
+): boolean {
+  for (const [field, values] of Object.entries(activeFilters)) {
+    const measurementValue = measurement[field];
+    if (values.length > 0 &&
+       ((typeof measurementValue === "string") && !values.includes(measurementValue))){
+      return false;
+    }
+  }
+  return true;
+}
+
+function createMeasurementsColoringData(
+  filters: MeasurementFilters,
+  groupBy: string,
+  groupingValue: string,
+  collection: Collection,
+): {
+  nodeAttrs: MeasurementsNodeAttrs,
+  colorings: Colorings,
+} {
+  const measurementColorBy = encodeMeasurementColorBy(groupingValue);
+  const activeMeasurementFilters = getActiveMeasurementFilters(filters);
+  const strainMeasurementValues: {[strain: string]: number[]} = collection.measurements
+    .filter((m) => m[groupBy] === groupingValue && matchesAllActiveFilters(m, activeMeasurementFilters))
+    .reduce((accum, m) => {
+      (accum[m.strain] = accum[m.strain] || []).push(m.value)
+      return accum
+    }, {});
+
+  const nodeAttrs: MeasurementsNodeAttrs = {};
+  for (const [strain, measurements] of Object.entries(strainMeasurementValues)) {
+    const averageMeasurementValue = measurements.reduce((sum, value) => sum + value) / measurements.length;
+    nodeAttrs[strain] = {
+      [measurementColorBy]: {
+        value: averageMeasurementValue
+      }
+    };
+  }
+  const sortedValues = collection.measurements
+    .map((m) => m.value)
+    .sort((a, b) => a - b);
+
+  // Matching the default coloring for continuous scales
+  const colorRange = colors[9];
+  const step = 1 / (colorRange.length - 1);
+  const measurementsColorScale: [number, string][] = colorRange.map((color, i) => {
+    return [quantile(sortedValues, (step * i)), color]
+  });
```

Comment:
> Are the bins for the color ramp for the measurements scale fixed across all measurements in a given collection?

Yes, the color scale is created using all measurements in a given collection. I did this to have a stable coloring scale for each collection.